Laravel 11 upgrade

kdmnk opened the pull request:
## Description
Updates dependencies, laravel 11, cleanup
## Related Issue
Closes #488 
## Changes
<!-- List the changes introduced by this pull request -->
- Updated dependencies
- Cleanup boilerplate files
- Skeleton for deployment docs
- Some extra refactoring with middlewares and gates
## TODO
<!-- Optional notes -->
- Invityr: https://github.com/EotvosCollegium/mars/pull/491


<!-- Please take a look at the Contributing guidelines and conventions before opening a PR -->

kdmnk committed:
update to Livewire 3

kdmnk committed:
cleanup boilerplate files

kdmnk committed:
update dependencies

kdmnk committed:
fix autogenerate warning

kdmnk committed:
deployment file skeleton

kdmnk commented on bootstrap/app.php:
```diff
@@ -1,55 +1,53 @@
+        $middleware->redirectGuestsTo('/login');
+        $middleware->trustProxies(
+            //TODO
+            //at: config('app.trust_proxies'),
```
@horcsinbalint I can't use config anymore to get the trusted proxies. Can you give me the list?

horcsinbalint replied:
Yep, it is `172.16.9.200`. The code supported lists but we only have a single reverse proxy distributing the work to the different webservices. (We might change it in the future but it should not be frequent and for this purpose we use private IP addresses so we have control over them.)

kdmnk committed:
add trusted proxies

kdmnk committed:
update phpunit.xml

kdmnk committed:
fix tests

deepsource-autofix[bot] committed:
style: format code with PHP CS Fixer

This commit fixes the style issues introduced in b0426ee according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/490

kdmnk committed:
added script for deployment

kdmnk committed:
Merge branch 'laravel11' of https://github.com/EotvosCollegium/mars into laravel11

kdmnk committed:
change ci pipeline order

kdmnk committed:
update mysql image for CI

kdmnk committed:
Revert "change ci pipeline order"

This reverts commit 5707fb4145f9d4b790bfa41bc380ed6439e1f05f.

kdmnk committed:
preset app key for ci

kdmnk committed:
Revert "preset app key for ci"

This reverts commit 0642ae63d2bc311249a4e62024c249717ae6cdd2.

kdmnk committed:
ping routers

kdmnk committed:
Fix missing docs

kdmnk committed:
added readme for livewire

kdmnk committed:
added missing docs

horcsinbalint commented on .github/workflows/ci.yml:
```diff
@@ -1,4 +1,4 @@
-on: [push, pull_request]
```
Is there a reason for disabling workflows if a PR is created? I had created #415 because of contributors who do not have write permission to the repo.

kdmnk replied:
Ah, I see. Otherwise in our case it's running twice but then I'll put it back.

kdmnk committed:
Split PossibleRedirects.php to NotifyAboutEvaluation.php and RedirectTenantsToUpdate.php

kdmnk committed:
Explicit argument name for isCollegist(alumni: ...)

kdmnk committed:
revert deleting pull_request from ci

deepsource-autofix[bot] committed:
style: format code with PHP CS Fixer

This commit fixes the style issues introduced in dce7607 according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/490

kdmnk committed:
add missing docs

kdmnk committed:
Merge branch 'development' into laravel11

kdmnk committed:
fix missing csrf token

kdmnk committed:
Merge branch 'laravel11' of https://github.com/EotvosCollegium/mars into laravel11

horcsinbalint commented on app/Http/Middleware/LogRequests.php:
```diff
@@ -0,0 +1,30 @@
+    public function handle(Request $request, Closure $next): Response
+    {
+        if (Auth::check() && $request->isMethod('post')) {
+            SystemLog::debug('User #' . user()->id . ' sent request: path='
```
Does this only log in debug mode or also in production?

kdmnk replied:
Same as before, the files just have been renamed. I think debug statements won't have an effect in production.

horcsinbalint replied:
I checked it, a DEBUG log is created and this middleware is active in production. But it is set in multiple routes not to invoke this middleware (which is great for things like someone votes on a general assembly, authentication etc.).

horcsinbalint commented on routes/web.php:
In line 267 `'log'` could also be changed to `LogRequests::class`

horcsinbalint replied:
Same for line 104

kdmnk replied:
Nice catch, thanks

kdmnk committed:
fix remaining 'log' middleware
